lib/gui/components/drive-selector/templates/drive-selector-modal.tpl.html

```diff
@@ -50,7 +50,17 @@ <h4 class="list-group-item-heading">{{ drive.description }} - {{ drive.size | gi
 </div>
 
 <div class="modal-footer">
-  <button class="button button-primary button-block"
+  <div class="modal-footer-warning"
+    ng-show="modal.state.hasDrive() && modal.state.isSystemDrive(modal.state.getDrive())">
```

Comment:
I like these ideas. Let's try the following:

- `.getButtonStatus()`: This function would return a string, like `primary`, `danger`, `disabled`, etc. The `ngDisabled` directive on the button can use it by comparing it to the string "disabled".

- `.getFooterMessage()`: This function could return `undefined`, or a string representing the message. It can even rely on `.getButtonStatus()` to see what it should show.

- (optional) `.hasFooterMessage()`: returns `true` if the result of `getFooterMessage()` is not `undefined`. This can used in an `ngShow` directive on the footer, just to make things nicer.

The nice thing about extracting all these functions to the controller is that we can unit test them all.